ddtrace/profiling/collector/_lock.py

```diff
@@ -221,6 +221,13 @@ def release(self, *args, **kwargs):
 
     acquire_lock = acquire
 
+    def __enter__(self):
+        self.acquire()
```

Comment:
Should we have a common helper with the acquire event logic to invoke here, and then `return self.__wrapped__.__enter__()`?

Reply:
Hmm, it looks like `__enter__` and `__exit__` are just different names of `acquire` and `release`. 
https://github.com/python/cpython/blob/7595e6743ac78ac0dd19418176f66d251668fafc/Modules/_threadmodule.c#L935C1-L938

I agree that it might be safer to do that when in future Python changes __enter__ and __exit__ to have more logic on top of `acquire` and `release`

Reply:
Done.